### DISCUSSION TITLE
Simple routing

### DISCUSSION THREAD

**Discussion by bubnenkoff** [static-web-server/static-web-server#234] [Q&A]
I like SWS, but it seems that it's only for serving static files.
But I need way to do simple routing. For example to redirect all requests to `/foo` to another service on my server.

Is it possible to do with SWS? Maybe there is any workaround?

**Comment by joseluisq** [comment]
> I like SWS, but it seems that it's only for serving static files.

SWS **is** designed for static file serving on purpose.

> But I need way to do simple routing. For example to redirect all requests to /foo to another service on my server.
> Is it possible to do with SWS? Maybe there is any workaround?

If you want to redirect requests within SWS or out of it (remote URLs) then you can use [URL Redirects](https://static-web-server.net/features/url-redirects/).

--

BTW If you have questions or need help then please use [the discussions](https://github.com/static-web-server/static-web-server/discussions) page instead of creating an issue.

**Reply by bubnenkoff** [reply]
@joseluisq thanks for response. But I can't understand if remote server is requesting `/foo` and I have got `URL Redirects` Rule will it be possible to send request back to remote server?

**Reply by joseluisq** [reply]
There are two ways to work with routes in SWS. You can either make just redirections (URL Redirect) or overwrite routes with custom ones (URL Rewrites).

- **[URL Redirects](https://static-web-server.net/features/url-redirects/)**  can just redirect from one place (route source) to another (route or remote URL destination). The condition here is that routes must point to an existing resource (either file in your system) or a remote URL.
- **[URL Rewrites](https://static-web-server.net/features/url-rewrites/)** can overwrite URL paths (routes), meaning that for example a route source (rewrite route) can or can not necessarily point to an existing resource (file in your system). So you could use it to create custom routes that then SWS will internally resolve to the underlying existing resource (file in your system).

Additionally, URL Rewrites will support [an additional redirection option](https://github.com/static-web-server/static-web-server/pull/231) to even redirect when using custom routes.

**Reply by joseluisq** [reply]
> But I can't understand if remote server is requesting /foo and I have got URL Redirects Rule will it be possible to send request back to remote server?

I think for your use case you probably want to use URL Rewrites so SWS will be able to reply with the requested file to your other server.

**Comment by joseluisq** [comment]
As mentioned in https://github.com/static-web-server/static-web-server/issues/228#issuecomment-1637487172.
The latest [v2.20.0](https://github.com/static-web-server/static-web-server/releases/tag/v2.20.0) already makes possible generic route rewrites with optional redirection. You can read more about it at https://static-web-server.net/features/url-rewrites/
